src/test/java/de/tum/in/www1/artemis/quiz/QuizExerciseIntegrationTest.java

```diff
@@ -104,199 +104,81 @@ class QuizExerciseIntegrationTest extends AbstractSpringIntegrationBambooBitbuck
-                assertThat(dropLocations.get(0).getHeight()).as("Height for drop location is correct").isEqualTo(10);
-                assertThat(dropLocations.get(1).getPosX()).as("Pos X for drop location is correct").isEqualTo(20);
-                assertThat(dropLocations.get(1).getPosY()).as("Pos Y for drop location is correct").isEqualTo(20);
-                assertThat(dropLocations.get(1).getWidth()).as("Width for drop location is correct").isEqualTo(10);
-                assertThat(dropLocations.get(1).getHeight()).as("Height for drop location is correct").isEqualTo(10);
-
-                List<DragItem> dragItems = dragAndDropQuestion.getDragItems();
-                assertThat(dragItems.get(0).getText()).as("Text for drag item is correct").isEqualTo("D1");
-                assertThat(dragItems.get(1).getText()).as("Text for drag item is correct").isEqualTo("D2");
-            }
-            else if (question instanceof ShortAnswerQuestion shortAnswerQuestion) {
-                assertThat(shortAnswerQuestion.getSpots()).as("Short answer question spots were saved").hasSize(2);
-                assertThat(shortAnswerQuestion.getSolutions()).as("Short answer question solutions were saved").hasSize(2);
-                assertThat(shortAnswerQuestion.getTitle()).as("Short answer question title is correct").isEqualTo("SA");
-                assertThat(shortAnswerQuestion.getText()).as("Short answer question text is correct").isEqualTo("This is a long answer text");
-                assertThat(shortAnswerQuestion.getPoints()).as("Short answer question score is correct").isEqualTo(2);
-
-                List<ShortAnswerSpot> spots = shortAnswerQuestion.getSpots();
-                assertThat(spots.get(0).getSpotNr()).as("Spot nr for spot is correct").isZero();
-                assertThat(spots.get(0).getWidth()).as("Width for spot is correct").isEqualTo(1);
-                assertThat(spots.get(1).getSpotNr()).as("Spot nr for spot is correct").isEqualTo(2);
-                assertThat(spots.get(1).getWidth()).as("Width for spot is correct").isEqualTo(2);
-
-                List<ShortAnswerSolution> solutions = shortAnswerQuestion.getSolutions();
-                assertThat(solutions.get(0).getText()).as("Text for solution is correct").isEqualTo("is");
-                assertThat(solutions.get(1).getText()).as("Text for solution is correct").isEqualTo("long");
-            }
-        }
+        createdQuizAssert();
     }
 
     @Test
     @WithMockUser(username = TEST_PREFIX + "instructor1", roles = "INSTRUCTOR")
     void createQuizExercise_setCourseAndExerciseGroup_badRequest() throws Exception {
-        ExerciseGroup exerciseGroup = database.addExerciseGroupWithExamAndCourse(true);
-        QuizExercise quizExercise = ModelFactory.generateQuizExerciseForExam(exerciseGroup);
+        ExerciseGroup exerciseGroup = database.createAndSaveActiveExerciseGroup(true);
+        quizExercise = ModelFactory.generateQuizExerciseForExam(exerciseGroup);
```

Comment:
Personally I would expect an attribute "quizExercise" to be initialized in a @BeforeAll and then used in the tests. For newly created objects I would expect a local variable.
What is your opinion?

Reply:
I didn't really consider that, but I think it makes sense. I will change it :+1: